daft/ml/_expressions.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from daft import Series
+    from daft.ml.protocols import TextEmbedder, TextEmbedderBatched
+    from daft.ml.typing import Embedding
+
+
+class _TextEmbedderExpression:
+    """Function expression implementation for a TextEmbedder protocol."""
+
+    _model: TextEmbedder
+
+    def __init__(self, model: TextEmbedder):
+        self._model = model
+        self._model.__enter__()  # initialize
```

Comment:
we also to "exit" the model as well? One option is to leverage https://docs.python.org/3/library/contextlib.html#contextlib.ExitStack
and call close in the __del__ method